### DISCUSSION TITLE
How can I override or augment the HTML page title for blog post pages?

### DISCUSSION THREAD

**Discussion by dsghi** [OrchardCMS/OrchardCore#11872] [Q&A] [answered]
Aloha, friends (apologies if this is a super noob question, I looked all over for an answer and couldn't come up with one)

I'm using the default page title format in the site settings:
`{% page_title Site.SiteName, position: "after", separator: " - " %}`
In my site layout.liquid I have `<title>{{ "PageTitle" | shape_new | shape_stringify }}</title>`
which gives me _page title_ - _site name_ as expected.

I've created a blog and, of course, have blog pages. The pages are in a list.
The situation is that on these blog pages, I'd like to modify the page title to be:
_page title_ - _blog name_ - _site name_

I've found that I can get the blog name with `{{ Model.ContentItem | container | display_text }}`. However, I can't figure out what to do to override the default site page format.

What would be the correct method for "overriding" the default page title to make a new one which includes the blog name for these pages?

**Comment by Skrypt** [comment, marked as answer]
https://docs.orchardcore.net/en/latest/docs/reference/modules/Liquid/#page_title

```liquid
{% assign page_title = "PageTitle" %}
{% assign blog_name = Model.ContentItem | container | display_text %}

{% page_title page_title, position: "after", separator: " - " %}
{% page_title blog_name, position: "after", separator: " - " %}
{% page_title Site.SiteName, position: "after", separator: " - " %}
```

This Liquid "Layout Tag" is additive to the current page title set in your layout.liquid file : 

```liquid
<title>{% page_title Site.SiteName, position: "before", separator: " - " %}</title>
```

See RenderTitleSegmentTag.cs for understanding how it works : 

https://github.com/OrchardCMS/OrchardCore/blob/f28f57198a709635f8e9b82957938343f8790070/src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Tags/RenderTitleSegmentsTag.cs#L1-L39

Alternatively, I think you can also do : 

```liquid
{% page_title "something" %}
```

Which should override the entire title if I remember correctly.

**Reply by dsghi** [reply]
Thank you, @Skrypt - I appreciate this detail. The key is that the tag is additive.

I got there using the following on my BlogPost page:
```
{% assign page_title = Model.ContentItem | display_text %}
{% assign blog_name = Model.ContentItem | container | display_text %}

{% page_title page_title, position: "after", separator: " - " %}
{% page_title blog_name, position: "after", separator: " - " %}
```
However, the challenge I face is that each of the `page_title` lines outputs text to the page. Any advice on preventing that?

To get by, I wrapped these in a hidden div, but that seems a little dirty. 🤣
```
<div class="d-none">
{% page_title page_title, position: "after", separator: " - " %}
{% page_title blog_name, position: "after", separator: " - " %}
</div>
```

**Reply by Skrypt** [reply]
Generally you should wrap this inside a `<h1 class="d-none"></h1>` for SEO purpose. Else an issue could be opened about having an option on the Tag helper to not render anything where it is called if that's possible.

**Reply by dsghi** [reply]
Oh, duh, yeah, I already have an H1 with the page title rendered; I'll change that over to use this output. Thanks again!
